configs/config-no-entropy.h

```diff
@@ -17,6 +17,9 @@
  * See README.txt for usage instructions.
  */
 
+#define MBEDTLS_PSA_CRYPTO_CONFIG_FILE "../configs/crypto-config-no-entropy.h"
+#define MBEDTLS_PSA_CRYPTO_CONFIG
```

Comment:
As we have done for other configuration files you need to enable MBEDTLS_PSA_CRYPTO_C and                                             
MBEDTLS_USE_PSA_CRYPTO which requires MBEDTLS_ENTROPY_C defeating the purpose of this configuration. I think we have some plans to be able to initialize and use only PSA APIs that do not rely on some entropy but in the mean time not sure what we should do with this configuration. @gilles-peskine-arm what do you think?

Reply:
We can have a PSA-based configuration without `entropy.c` by using `MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG`, which is a simulated entropy source (encapsulated in a simulated RNG). But we already have components with that option, and it's not the goal here. The goal here is to enable the parts of the library that don't require entropy.

In the future, we'll have a way to make PSA crypto work without a RNG, which will allow to do the same things that `config-no-entropy.h` currently does (hash, MAC, symmetric ciphers, KDF, signature verification, X.509 verification). But that requires work on splitting `psa_crypto_init`, and we haven't done that work yet (we were thinking of doing it in Q3, but I suspect it'll get delayed to Q4 so we can focus on the repo split).

For now, we can either keep `config-no-entropy.h` as is, or remove it but file an issue to re-create it when we can, i.e. as part of the split-init epic. I think re-creating it later is the way to go, because keeping legacy config options for some time would hurt other work.